File: ED61/b.rb
if File.exists?("input")
  $stdin = File.open("input")
  $stdout = File.open("output","w")
end

n = gets.to_i
arr = gets.chomp.split(" ").map(&:to_i).sort
sum = arr.inject(:+)
m = gets.to_i
qarr = gets.chomp.split(" ").map(&:to_i)

qarr.each do |q|
	puts sum-arr[-q]
end

File: ED61/b.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main(){
	
	#ifndef ONLINE_JUDGE
    	freopen("input", "r", stdin);
    	freopen("output", "w", stdout);
	#endif

    int n;
    cin>>n;
    ll arr[n];    
    ll sum=0;
    for(int i=0;i<n;i++){
    	cin>>arr[i];
    	sum+=arr[i];
    }

    sort(arr,arr+n);

    ll m,k;
    cin>>m;

    for(int i=0;i<m;i++){
    	cin>>k;
    	cout<<sum-arr[n-k]<<endl;
    }
}